crates/sha512/src/sha512.rs

```diff
@@ -152,10 +164,50 @@ where
     Ok(cur.into_iter().flat_map(|e| e.into_bits_be()).collect())
 }
 
+pub fn sha512_256t<Scalar, CS>(
+    mut cs: CS,
+    input: &[Boolean],
+) -> Result<Vec<Boolean>, SynthesisError>
+where
+    Scalar: PrimeField,
+    CS: ConstraintSystem<Scalar>,
+{
+    assert!(input.len() % 8 == 0);
+
+    let mut padded = input.to_vec();
+    let plen = padded.len() as u128;
+    // append a single '1' bit
+    padded.push(Boolean::constant(true));
+    // append K '0' bits, where K is the minimum number >= 0 such that L + 1 + K + 128 is a multiple of 1024
+    while (padded.len() + 128) % 1024 != 0 {
+        padded.push(Boolean::constant(false));
+    }
+    // append L as a 128-bit big-endian integer, making the total post-processed length a multiple of 1024 bits
+    for b in (0..128).rev().map(|i| (plen >> i) & 1 == 1) {
+        padded.push(Boolean::constant(b));
+    }
+    assert!(padded.len() % 1024 == 0);
```

Comment:
This padding code block is a repeat of the padding logic in `sha512`. You could consider having a helper function which adds this padding in both `sha512` and `sha512_256t`. May be overkill though.

Reply:
You're right. I've extended the PR to support purely truncated SHA-512 (which uses the same IVs as SHA-512, contrarily to SHA-512/256), and mutualizing the padding block was useful to minimize boilerplate.